### ISSUE TITLE
Typescript suddenly incompatible between SST and CDK constructs

### ISSUE THREAD

**Issue by louismorgan-linaro** [sst/sst#3552]
I'm suddenly getting typescript issues when passing CDK constructs into SST construct fields.

e.g. passing a `PolicyStatement` into an api handler's `initialPolicy`
```
Type 'import("/home/louis/ldp/node_modules/aws-cdk-lib/aws-iam/lib/policy-statement").PolicyStatement' is not assignable to type 'import("/home/louis/ldp/node_modules/sst/node_modules/aws-cdk-lib/aws-iam/lib/policy-statement").PolicyStatement'.
  Types have separate declarations of a private property '_action'.ts(2322)
```
I'm having similar errors with pretty much all cases where CDK constructs are passed to SST

**Comment by jayair** [comment]
Make sure your CDK versions match what `sst version` prints?

**Comment by splicebison** [comment]
@jayair @louismorgan-linaro 

This is due to `aws-cdk-lib` being installed twice with two different versions. Scoped `@aws-cdk/*` libs have a dependency on `^2.110.1` whereas the latest version installed is `2.114.0`. This is what is causing the conflict.

You can temporarily work around this by setting your resolutions in your package.json file as such:

```json
"resolutions": {
  "aws-cdk-lib": "2.110.1"
},
``` 

After, install again and the TS conflicts will go away.

**Comment by louismorgan-linaro** [comment]
I managed to fix it yesterday by just uninstalling `aws-cdk-lib`, `constructs` and `sst` then reinstalling `sst` only. It all seems to work but I think when I initiated the repo with `create-sst` it had `aws-cdk-lib` explicitly in the `package.json` file.

Is there any reason why I shouldn't proceed with only `sst` as an explicit dependency?

**Comment by splicebison** [comment]
> Is there any reason why I shouldn't proceed with only sst as an explicit dependency?

Yes, a few. It's unreliable to others on your team as you're not explicitly setting project dependencies (even if you are solo, please make this a good habit). You are also relying on peer dependencies for app/development functionality. If those peers change one day, it will break your apps.

By using the `resolutions` prop, it allows you to continue to (correctly) ever-green your project dependencies until peers / other libs catch-up or fix their dependency tree. The resolution can be removed at a later date when it's fixed by the package maintainers. Also, this way, you are then explicitly declaring a dev dependency for dev work that is required within your project.

**Comment by louismorgan-linaro** [comment]
Thanks very much for all the info @splicebison !
